dwave/cloud/utils.py

```diff
@@ -867,11 +873,52 @@ def get_contrib_packages():
 def get_platform_tags():
     """Return a list of platform tags generated from registered entry points."""
 
-    fs = [ep.load() for ep in iter_entry_points('dwave.common.platform.tags')]
+    fs = [ep.load() for ep in entry_points(group='dwave.common.platform.tags')]
     tags = list(filter(None, [f() for f in fs]))
     return tags
 
 
+class VersionNotFoundError(Exception):
+    """Package version requirement not satisfied."""
+
+
+def get_distribution(requirement: Union[str, Requirement],
```

Comment:
Will [this functionality](https://github.com/dwavesystems/dwave-ocean-sdk/blob/fe16c75bf77583b78f6661754d1b1048b3704050/docs/conf.py#L213) be affected and should it be updated ([this PR](https://github.com/dwavesystems/dwave-ocean-sdk/pull/274) moves the doc build to py3.10)?

Reply:
That functionality is affected by the fact that Python 3.12 does not create virtual environments with `setuptools` preinstalled. `pkg_resources` is provided by `setuptools`. And that's the reason for some of these changes here - to support default py312 environments. One (lazy) alternative is to explicitly install `setuptools`. Another option is to switch to `importlib.resources` and `importlib.metadata`, depending on your requirements.

The way you use `pkg_resources.get_distribution` in Ocean's docs `conf.py` can easily be replaced with [`importlib.metadata.distribution`](https://docs.python.org/3/library/importlib.metadata.html#distributions).